### PR TITLE
Make Magician and Pickpocket have a chance to proc to match Covet and other itemsteal mechanics

### PR DESCRIPTION
**Pull request by Fontbane**
As it is, fighting a boss Meowscarada or anything else with Magician is just like going against Mini Black Hole. Having items stolen every hit with Magician is inconsistent with other item stealing mechanics implemented with the game's unique item system in mind, like Covet and Grip Claw. So, this PR gives Magician and Pickpocket a chance to proc, rather than being guaranteed. Magician has a 1/3 chance to steal, and Pickpocket has a 1/2 chance. Pickpocket has a higher chance because there's at least a way to avoid it, by not using contact moves. These numbers can be adjusted if need be.

### CODE REVIEW

**Comment by OrangeRed** [comment]
Rather than hard coding these values into the class can you add them as constructor params and use those instead
like this
```ts
constructor(condition?: PokemonAttackCondition, procChance: number) {
  super()

  this.condition = condition
  this.procChance = procChance
}
```

That way the proc chance is tied to the individual ability

**Comment by Fontbane** [comment]
> Rather than hard coding these values into the class can you add them as constructor params and use those instead like this
> 
> ```ts
> constructor(condition?: PokemonAttackCondition, procChance: number) {
>   super()
> 
>   this.condition = condition
>   this.procChance = procChance
> }
> ```
> 
> That way the proc chance is tied to the individual ability

I considered this tbh, I think it would be cleaner if I did actually.

File: src/data/ability.ts [hunk 1]
```diff
@@ -1224,18 +1224,20 @@ export class PostAttackApplyBattlerTagAbAttr extends PostAttackAbAttr {
 
 export class PostDefendStealHeldItemAbAttr extends PostDefendAbAttr {
   private condition: PokemonDefendCondition;
+  private chance: integer; //Proc chance out of 100
 
-  constructor(condition?: PokemonDefendCondition) {
+  constructor(condition?: PokemonDefendCondition, chance?: integer) {
```
Comment (OrangeRed) [inline src/data/ability.ts:1229]: I'm pretty sure we use fractions and decimals everywhere for probability. Can you make this consistent with that?

Also we recently started asking everyone to write jsdoc comments so could you do that too?

Comment (Fontbane) [reply src/data/ability.ts:1229]: Actually we use integers for all the other chance parameters for ability attributes. I'll add the comments though.

**Comment by brain-frog** [comment]
I appreciate the added documentation.I do not see a reason to add probability to an ability that is 100% chance normally.

**Pull request closed without merging**